handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -284,6 +284,15 @@ public SslHandler(SSLEngine engine, boolean startTls, Executor delegatedTaskExec
 
         wantsDirectBuffer = engine instanceof OpenSslEngine;
         wantsLargeOutboundNetworkBuffer = !(engine instanceof OpenSslEngine);
+
+        /**
+         * As JDK {@link SSLEngine#unwrap(ByteBuffer, ByteBuffer)} only works with one {@link ByteBuffer} we should not
+         * try to use a {@link CompositeByteBuf} at the first place to make it as fast as possible.
+         *
+         * When using {@link OpenSslEngine} we can call {@link OpenSslEngine#unwrap(ByteBuffer[], ByteBuffer[])} and
+         * so remove the need of memory copies for more performance.
+         */
+        setCumulator(engine instanceof OpenSslEngine ? COMPOSITE_CUMULATOR : MERGE_CUMULATOR);
```

Comment:
We do `engine instanceof OpenSslEngine` three times in this block. Let's just extract a variable.